src/backend/api/reservations.js

```diff
@@ -0,0 +1,91 @@
+const express = require("express");
+const router = express.Router();
+const knex = require("../database");
+
+// GET return all reservations
+
+router.get("/", async (request, response) => {
+    try {
+      const reservations = await knex("Reservation").select("*");
+      response.json(reservations);
+    } catch (error) {
+      response.status(500).json({ error: "Error retrieving reservations" });
+    }
+  });
+
+
+// POST Adds a new reservation to the database ??
+
+router.post("/", async (request, response) => {
+    const newReservation = request.body;
+    newReservation.created_date = new Date();
```

Comment:
This is getting into the more advanced realm, but for cases like this where you need to modify an _input parameter_ (here, the request body), it is better to create a clone of the object (here, the body) and do the modifications there.

So I would rewrite it like this :)

```javascript
const newReservation = {
    ...request.body,
    created_date: new Date()
};
```